### PR TITLE
[BEAM-2411] Make the write transform of HBaseIO simpler

### PR DESCRIPTION
**Pull request by iemejia**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`.
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by iemejia** [comment]
R: @jkff A simplification based on our discussion on the Bigtable-HBase duality.

**Comment by iemejia** [comment]
There is an extra thing, notice that I made elementCounters for debug per bundle, this is a fix on the previous version, and aligns it with BigtableIO.

File: sdks/java/io/hbase/src/main/java/org/apache/beam/sdk/io/hbase/HBaseIO.java [hunk 1]
```diff
@@ -679,6 +667,5 @@ public void populateDisplayData(DisplayData.Builder builder) {
         }
     }
 
-    public static final Coder<KV<byte[], Iterable<Mutation>>> WRITE_CODER =
-            KvCoder.of(ByteArrayCoder.of(), IterableCoder.of(HBaseMutationCoder.of()));
+    public static final Coder<Mutation> WRITE_CODER = HBaseMutationCoder.of();
```
Comment (jkff) [inline sdks/java/io/hbase/src/main/java/org/apache/beam/sdk/io/hbase/HBaseIO.java]: While we're at simplifying things.. I believe you can remove this and all the setCoder calls by using something like https://github.com/apache/beam/blob/master/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryCoderProviderRegistrar.java

**Comment by iemejia** [comment]
Thanks for the hint on the CoderProviderRegistrar, I am going to take a look and add it to the PR

**Comment by lukecwik** [comment]
Retest this please

**Comment by iemejia** [comment]
Thanks for the review/ideas Eugene.

**Pull request merged**